File: salesman-android/android/app/src/main/java/com/hometech/app/BootReceiver.java
package com.hometech.app;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Build;
import android.util.Log;

import static com.hometech.app.LocationForegroundService.KEY_ACTIVE;
import static com.hometech.app.LocationForegroundService.PREFS_NAME;

public class BootReceiver extends BroadcastReceiver {
    private static final String TAG = "HomeTechBootReceiver";

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent == null || intent.getAction() == null) return;

        String action = intent.getAction();
        boolean bootCompleted =
            Intent.ACTION_BOOT_COMPLETED.equals(action) ||
            "android.intent.action.QUICKBOOT_POWERON".equals(action);
        if (!bootCompleted) return;

        SharedPreferences prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
        if (!prefs.getBoolean(KEY_ACTIVE, false)) return;

        Intent serviceIntent = new Intent(context, LocationForegroundService.class);
        serviceIntent.setAction(LocationForegroundService.ACTION_START);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(serviceIntent);
        } else {
            context.startService(serviceIntent);
        }
        Log.i(TAG, "Restarted active location tracking after boot");
    }
}

File: android-app/app/src/main/java/com/hometech/app/BootReceiver.kt
package com.hometech.app

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import androidx.core.content.ContextCompat

/**
 * Restarts duty tracking after device reboot or app update,
 * but only if the salesman had an active duty (token persisted in prefs).
 */
class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action ?: return
        if (action != Intent.ACTION_BOOT_COMPLETED &&
            action != Intent.ACTION_MY_PACKAGE_REPLACED &&
            action != "android.intent.action.QUICKBOOT_POWERON") return

        val prefs = context.getSharedPreferences(
            LocationTrackingService.PREFS_NAME, Context.MODE_PRIVATE
        )
        val token = prefs.getString(LocationTrackingService.KEY_TOKEN, "") ?: ""
        val base  = prefs.getString(LocationTrackingService.KEY_BASE_URL, "") ?: ""
        if (!LocationTrackingService.hasActiveDuty(prefs) ||
            token.isEmpty() || base.isEmpty()) return

        Log.i("HomeTechTracking", "BootReceiver → resuming duty tracking")
        val svc = Intent(context, LocationTrackingService::class.java).apply {
            setAction(LocationTrackingService.ACTION_START)
        }
        try {
            ContextCompat.startForegroundService(context, svc)
        } catch (e: Exception) {
            Log.w("HomeTechTracking", "BootReceiver failed: ${e.message}")
        }
    }
}
